bindings/alicloud/sls/sls.go

```diff
@@ -0,0 +1,127 @@
+package sls
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	sls "github.com/aliyun/aliyun-log-go-sdk"
+	"github.com/aliyun/aliyun-log-go-sdk/producer"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+)
+
+type AliCloudSlsLogstorage struct {
+	logger   logger.Logger
+	producer *producer.Producer
+	metadata SlsLogstorageMetadata
+}
+
+type SlsLogstorageMetadata struct {
+	Endpoint        string `json:"endpoint"`
+	AccessKeyID     string `json:"accessKeyID"`
+	AccessKeySecret string `json:"accessKeySecret"`
+}
+
+// parse metadata field
+func (s *AliCloudSlsLogstorage) Init(metadata bindings.Metadata) error {
+	m, err := s.parseMeta(metadata)
+	if err != nil {
+		return err
+	}
+	s.metadata = *m
+	producerConfig := producer.GetDefaultProducerConfig()
+	// the config properties in the component yaml file
+	producerConfig.Endpoint = m.Endpoint
+	producerConfig.AccessKeyID = m.AccessKeyID
+	producerConfig.AccessKeySecret = m.AccessKeySecret
+	s.producer = producer.InitProducer(producerConfig)
+	return nil
+}
+
+func NewAliCloudSlsLogstorage(logger logger.Logger) *AliCloudSlsLogstorage {
+	logger.Debug("new a Slslog component object")
+	s := &AliCloudSlsLogstorage{
+		logger: logger,
+	}
+	return s
+}
+
+func (s *AliCloudSlsLogstorage) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	// verify the metadata property
+	if logProject := req.Metadata["project"]; logProject == "" {
+		return nil, fmt.Errorf("SLS binding error: project property not supplied")
+	}
+	if logstore := req.Metadata["logstore"]; logstore == "" {
+		return nil, fmt.Errorf("SLS binding error: logstore property not supplied")
+	}
+	if topic := req.Metadata["topic"]; topic == "" {
+		return nil, fmt.Errorf("SLS binding error: topic property not supplied")
+	}
+	if source := req.Metadata["source"]; source == "" {
+		return nil, fmt.Errorf("SLS binding error: source property not supplied")
+	}
+
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, syscall.SIGTERM, os.Interrupt)
+
+	// Start producer instancce
+	s.producer.Start()
+	log, err := s.parseLog(req)
+	if err != nil {
+		s.logger.Info(err)
+		return nil, err
+	}
+
```

Comment:
I think it would be better to check explicitly that request only is for supported Operation (currently CreateOperation), else throw UnsupportedOperation error

Reply:
there is standard verify func int the runtime,so may be it’s not necessary at here，I think

`if binding, ok := a.outputBindings[name]; ok {
		ops := binding.Operations()
		for _, o := range ops {
			if o == req.Operation {
				var resp *bindings.InvokeResponse
				policy := a.resiliency.ComponentOutboundPolicy(a.ctx, name)
				err := policy(func(ctx context.Context) (err error) {
					resp, err = binding.Invoke(ctx, req)
					return err
				})
				return resp, err
			}
		}
		supported := make([]string, 0, len(ops))
		for _, o := range ops {
			supported = append(supported, string(o))
		}
		return nil, errors.Errorf("binding %s does not support operation %s. supported operations:%s", name, req.Operation, strings.Join(supported, " "))
	}`

Reply:
I did that before, but it didn't work. get the nonsupport error before invoke

Reply:
Ok, I get that. There are several examples like following, where-in explicitly check is done: https://github.com/dapr/components-contrib/blob/d38c7867715fd5d2d0d788fa440b22dd8440e49f/bindings/alicloud/rocketmq/rocketmq.go#L204

But, runtime should be able to take care of this.